django/contrib/redirects/migrations/0002_add_domain.py

```diff
@@ -0,0 +1,40 @@
+from django.apps import apps as django_apps
+from django.core.exceptions import ImproperlyConfigured
+from django.db import migrations, models
+
+
+def check_redirects_data(apps, schema_editor):
+    db_alias = schema_editor.connection.alias
+
+    Redirect = apps.get_model('redirects', 'Redirect')
+
+    if (Redirect.objects.using(db_alias).exists() and
+            not django_apps.is_installed('django.contrib.sites')):
+        raise ImproperlyConfigured(
```

Comment:
Why does it need to be installed in this migration? This migration doesn't even do anything. It's not given that all new migrations are applied at the same time.

Reply:
It's here to prevent schema change by AddField and possible data loss. 

Technically you're correct that migrations can be applied separately, but documentation explicitly says 'Run python manage.py migrate'. Also, the app cannot be considered working until all migrations are applied anyway.

Reply:
How is the schema change related to data loss?
I would recommend to have those checks in the next migration, the data migration.